Assets/Shopify/UIToolkit/Internal/ShopCredentials.cs

```diff
@@ -0,0 +1,51 @@
+﻿using UnityEngine;
+
+namespace Shopify.UIToolkit {
+    [System.Serializable]
+    public class ShopCredentials {
+        public delegate void UpdateHandler(ShopCredentials self);
+        public event UpdateHandler OnUpdate;
```

Comment:
This is an observer pattern so that we can observe the changes to this class and invalidate the client... i dont love it :/